### PR TITLE
Issue 105

### PR DESCRIPTION
**Pull request by axlpado**

### CODE REVIEW

File: modules/common/src/main/scala/notebook/front/widgets/package.scala [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import play.api.libs.json._
-
-import notebook._, front._, widgets._
-import notebook._, JSBus._
+import notebook._
+import notebook._
 import JsonCodec._
+import scala.util.Random
+import play.api.libs.json.Json.JsValueWrapper
+import play.api.libs.json.Json.JsValueWrapper
```
Comment (andypetrella) [inline modules/common/src/main/scala/notebook/front/widgets/package.scala]: not a drame, but some duplicates
```scala
import notebook._
import notebook._
```
and
```scala
import play.api.libs.json.Json.JsValueWrapper
import play.api.libs.json.Json.JsValueWrapper
```

File: modules/common/src/main/scala/notebook/front/widgets/package.scala [hunk 2]
```diff
@@ -127,15 +128,160 @@ package object widgets {
...
+    
+  }
+  
+  def lineChart(width: Int, jsons: Seq[(String, Any)]) = {
+    val id = Math.abs(Random.nextInt).toString
+    <div>
+    	<svg id={ "line"+id } width="600px" height="400px"
+       xmlns="http://www.w3.org/2000/svg" version="1.1">
+    	{
+      scopedScript(
```
Comment (andypetrella) [inline modules/common/src/main/scala/notebook/front/widgets/package.scala]: We can try to externalize this,  but if it's work I'm more than happy.
If possible using the injected deps would be better, but I know it's always tricky is Js
I'll take a look my self when merged ;-)

File: modules/common/src/main/scala/notebook/util/Reflector.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package notebook.util
+
+import play.api.libs.json.Json
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+import play.api.libs.json.JsNumber
+import play.api.libs.json.JsBoolean
+import play.api.libs.json.Json.JsValueWrapper
+
+object Reflector{
```
Comment (andypetrella) [inline modules/common/src/main/scala/notebook/util/Reflector.scala]: It's as simple as it's awesome at use!
I like that one

**Comment by andypetrella** [comment]
Besides some minor stuffs and some generalization we'll can do, this PR is just gold.

So I'll give it a try tomorrow (tired now) before merging it!

Thanks @axlpado !

**Comment by andypetrella** [comment]
Yeeehaaaa. Small stuffs here and there. It's so neat and great to have that, thx @axlpado

**Comment by axlpado** [comment]
Oh sorry, I missed the thread.
Yeah it has to be refined for sure.

Speaking about injected deps.... I put all the code in package.scala because I saw that html generation was there. Did you think about moving html generation in play templates instead of using scala xml dsl ? it could bring more code separation.

Happy to see it live :-D

**Comment by andypetrella** [comment]
No problemo. It's okay it works. 
I tend to make things evolving based on experiences and needs. So we'll see at usage how it could be refactored or improved, if any!
I prefer to see features live than having them stalling until an ideal world comes up 😅

**Pull request merged**